cmd/helm/helm.go

```diff
@@ -92,6 +92,11 @@ func initActionConfig(actionConfig *action.Configuration, allNamespaces bool) {
 		// TODO return error
 		log.Fatal(err)
 	}
+	err = kc.IsReachable()
```

Comment:
Two things come to mind here...

1. #6232 is in flight to remove the `init` command. Right now it only setups up missing files locally as tiller is gone. The idea is to lazy create the directories if not present.
2. The init action no longer requires communicating with the server. In this function the goal is to get the client set which is locally available and the namespace which is also locally available. If I'm on a plane working on something and need to init helm why shouldn't it work? This is a real work situation for me a couple times a year. Maybe others.

Does it matter if the Kubernetes cluster is available at this step?

Reply:
I second point #2:
- there are times where I don't have a network and still use some parts of helm for development.
- there are times when I have internet but can't reach my cluster (VPN not activated).  

I guess verifying the cluster connection should be done for commands that actually need the cluster.

For example, `helm --help` should always work (it does not with this patch).
`helm repo add ...`, `helm show ...` are other examples where the cluster is not needed.

Reply:
It only makes sense to have the check run for certain commands which would use kubernetes. Thank you for helping me out. I have updated the PR accordingly.

Reply:
@mattfarina 
we should have way to check cluster connection just to make sure helm commands going to work. How about having explicit helm command to check such as `helm cluster status` ? 
cc: @marckhouzam

Reply:
@mattfarina @jdolitsky I have made further changes to the PR. Have added the check only in places which seem necessary in terms of connecting to the Kubernetes cluster.